LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/JoinRequestMessageHandler.cs

```diff
@@ -57,6 +58,7 @@ internal async Task ProcessJoinRequestAsync(LoRaRequest request)
             try
             {
                 var timeWatcher = request.GetTimeWatcher();
+                using var joinAcceptCancellationToken = new CancellationTokenSource(timeWatcher.InTimeForJoinAccept() ? timeWatcher.GetRemainingTimeToJoinAcceptSecondWindow() : TimeSpan.Zero);
```

Comment:
The cts will be canceled immediately, if we are already outside the join accept window. I think that's ok, but we should make sure the behavior is not changed. Previously, for missing the window, we log, but we do not throw. This would now cause the request to immediately throw and that exception would then abort the request, but logged as an error in the system. 

I would suggest to handle that the same way we used to. If we get the OperationCanceledException that was triggered by our TCS we should handle it the same way as when we missed the window.

Reply:
to avoid 2 calculations (InTimeForJoinAccept and GetRemaining...) you could get the remaining time and make sure it's not negative. Also, we may want to add some margin to it to allow for cleanup. If we deduct ~80ms we should be able to stop the processing before the next join comes in.

Reply:
As discussed offline, no throwing will happen.

I fixed the catched exception in 711c8ee and the timeout in b5208f6 and in b5208f6